### PR TITLE
Web server, for discussion only

### PR DESCRIPTION
**Pull request by rakoo**
I open this PR for discussion, I don't expect it to be merged as-is.

Fixes #60

### CODE REVIEW

File: cmd/restic/cmd_web.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+		if !found {
+			return nil, fmt.Errorf("%s not found", name)
+		}
+		children, err := object.(childrener).loadChildren()
+		if err != nil {
+			return nil, err
+		}
+		for _, child := range children {
+			if _, ok := hr.objects[child.Name()]; !ok {
+				hr.objects[child.Name()] = child
```
Comment (rakoo) [inline cmd/restic/cmd_web.go]: High risk of collision here: we only use the node's name, we should use its full path

**Comment by fd0** [comment]
Thanks for your contribution. This already looks really good, although the web pages aren't really conforming to HTML ;)

**Comment by rakoo** [comment]
I'd like to keep using go's http stuff here, and maybe ask upstream for a better dirList function (the one that turns a list of entries to HTML).

**Comment by fd0** [comment]
Hm, it doesn't look hard to implement this with a template...

**Comment by fd0** [comment]
Hey, what's the status of this pull request, what are your plans?

In my opinion, we should just render the directory listing ourselves, and not depend on the stdlib stuff. It's not that hard...

**Comment by fd0** [comment]
@rakoo ping

I'd like to include the webserver in the first release, do you think we can get this in a usable state until Thursday?

**Comment by rakoo** [comment]
Hey @fd0 I'm really sorry but I can't work on it at the moment.

**Comment by fd0** [comment]
Okay, no problem. Shall we suspend the PR until you have time, or should I continue to work on it?

**Comment by xor-gate** [comment]
I have tried this on current master, but alot of internal API changes have been submitted. I think this pull request should be closed until we have a new web frontend which is mergable.

**Comment by fd0** [comment]
I agree, too much has changed, so I'll close this PR for now.

**Pull request closed without merging**